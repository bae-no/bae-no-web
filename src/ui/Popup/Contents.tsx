import { Fragment } from "react";

import {
  AlertDialogContentProps,
  Content,
  Overlay,
  Portal,
} from "@radix-ui/react-alert-dialog";

import { useMounted } from "src/hooks/useMounted";

import { overlayCss } from "../modalBase.css";

import { popupContentCss } from "./Popup.css";

export const AlertDialogContent = ({
  children,
  ...props
}: AlertDialogContentProps) => {
  const mount = useMounted();
  const Container = mount ? Portal : Fragment;

  return (
    <Container>
      <Overlay className={overlayCss} />
      <Content className={popupContentCss} {...props}>
        {children}
      </Content>
    </Container>
  );
};
