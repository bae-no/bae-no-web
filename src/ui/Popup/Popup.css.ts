import { style } from "@vanilla-extract/css";

import { contentCss } from "../modalBase.css";
import { sprinkles } from "../sprinkles.css";

export const popupContentCss = style([
  {
    width: "31.1rem",
  },
  contentCss,
  sprinkles({ bg: "white", br: "12", p: "24", position: "fixed" }),
]);
