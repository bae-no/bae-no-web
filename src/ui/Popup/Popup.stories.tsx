import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Button } from "../Button";
import { colors } from "../tokens/color";

import Popup from "./Popup";

export default {
  argTypes: {
    color: { control: "select", options: Object.keys(colors) },
  },
  component: Popup,
  title: "UI/Popup",
} as ComponentMeta<typeof Popup>;

const Template: ComponentStory<typeof Popup> = (args) => <Popup {...args} />;

export const Default = Template.bind({});
Default.args = {
  cancelText: "tesaaaa",
  children: <Button>팝업열기</Button>,
  confirmText: "test",
  description: "fdsa",
  title: "fas",
};
