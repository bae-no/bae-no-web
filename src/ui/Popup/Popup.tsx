import { ReactNode } from "react";

import {
  Portal,
  Overlay,
  Content,
  AlertDialogContentProps,
  AlertDialog,
  AlertDialogProps,
  AlertDialogTrigger,
  AlertDialogTitle,
  AlertDialogDescription,
  AlertDialogCancel,
  AlertDialogAction,
} from "@radix-ui/react-alert-dialog";

import { Box } from "../Box";
import { Button } from "../Button";
import { overlayCss } from "../modalBase.css";
import { Typography } from "../Typography";

import { popupContentCss } from "./Popup.css";

const AlertDialogContent = ({
  children,
  ...props
}: AlertDialogContentProps) => (
  <Portal>
    <Overlay className={overlayCss} />
    <Content className={popupContentCss} {...props}>
      {children}
    </Content>
  </Portal>
);

interface PopupProps extends Pick<AlertDialogProps, "open" | "onOpenChange"> {
  buttonDirection?: "row" | "column";
  cancelText?: string;
  children?: ReactNode;
  confirmText: string;
  description: string;
  onCancel?: () => void;
  onConfirm?: () => void;
  title: string;
}

const Popup = ({
  children,
  title,
  description,
  confirmText,
  cancelText,
  onConfirm,
  onCancel,
  buttonDirection = "row",
  open,
  onOpenChange,
}: PopupProps) => (
  <AlertDialog open={open} onOpenChange={onOpenChange}>
    {children && <AlertDialogTrigger asChild>{children}</AlertDialogTrigger>}
    <AlertDialogContent>
      <Box gap={buttonDirection === "column" ? "32" : "24"}>
        <Box alignItems="center" gap="8">
          <AlertDialogTitle asChild>
            <Typography as="h2" fontSize="headline4">
              {title}
            </Typography>
          </AlertDialogTitle>
          <AlertDialogDescription asChild>
            <Typography color="black3" fontSize="body1-m">
              {description}
            </Typography>
          </AlertDialogDescription>
        </Box>
        <Box
          flexDirection={
            buttonDirection === "column" ? "column-reverse" : "row"
          }
          gap={buttonDirection === "column" ? "8" : undefined}
        >
          {cancelText && (
            <AlertDialogCancel asChild>
              <Button color="white" onClick={onCancel}>
                {cancelText}
              </Button>
            </AlertDialogCancel>
          )}
          <AlertDialogAction asChild>
            <Button type="submit" onClick={onConfirm}>
              {confirmText}
            </Button>
          </AlertDialogAction>
        </Box>
      </Box>
    </AlertDialogContent>
  </AlertDialog>
);

export default Popup;
