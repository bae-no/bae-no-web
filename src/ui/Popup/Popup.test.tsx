import { render, screen } from "@testing-library/react";
import Popup from "./Popup";

describe("UI Popup Component", () => {
  it("match snapshot", () => {
    const { container } = render(
      <Popup
        cancelText="tesaaaa"
        confirmText="test"
        description="fdsa"
        buttonDirection="column"
        title="fas"
      >
        <div>test</div>
      </Popup>
    );
    expect(container).toMatchSnapshot();
  });

  it("render text", () => {
    render(
      <Popup
        cancelText="tesaaaa"
        confirmText="test"
        description="fdsa"
        buttonDirection="column"
        title="fas"
      >
        <div>tet</div>
      </Popup>
    );

    expect(screen.getByText("tet")).toBeInTheDocument();
  });
});
