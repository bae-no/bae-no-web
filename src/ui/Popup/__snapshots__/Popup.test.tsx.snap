// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Popup Component match snapshot 1`] = `
<div>
  <div
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-state="closed"
    type="button"
  >
    test
  </div>
</div>
`;
